### ISSUE TITLE
Passing Agruments 

### ISSUE THREAD

**Issue by hieubui2409** [Milad-Akarie/auto_route_library#1154]
I have wrapped route like that:
![image](https://user-images.githubusercontent.com/95897872/179339883-ea2a9e72-099e-4dc9-a43b-db97722659b1.png)
My setup in router:
![image](https://user-images.githubusercontent.com/95897872/179339821-e2bb79c7-ddcc-4023-8366-218741bc89fb.png)
In Original Page widget
![image](https://user-images.githubusercontent.com/95897872/179339787-48f4ded1-778d-41a0-bb76-093860d8d10e.png)
Your library generated:
![image](https://user-images.githubusercontent.com/95897872/179339800-c2486b4a-b0ab-4975-99d3-1776fa903813.png)
![image](https://user-images.githubusercontent.com/95897872/179339805-cd52d6ca-0c3f-45fc-be31-a5854d4092a5.png)
It throw the error:
![image](https://user-images.githubusercontent.com/95897872/179339811-f3a25f2d-8af6-415e-b82d-ce5b71b7ef34.png)
Please tell me if I did anythn=ing wrong. I'm new to Flutter

**Comment by Milad-Akarie** [comment]
@hieubui2409 are you navigating using paths?

**Comment by hieubui2409** [comment]
No, I navigate using `context.router.push(const UserPageWrapperRoute());`

**Comment by hieubui2409** [comment]
@Milad-Akarie I realized that, if i have tab navigation using TabAutoRouter.tabBar() or using wrapper route like above and I want to pass arguments to child tab / wrapped route, the first one which mark `initial` will nerver receive arguments and if i mark one fields of args as `required`, it will throw erro like `*** can not be null because it has a required parametter`. Please support me. Thanks in advance.

**Comment by Milad-Akarie** [comment]
Thia happened because your initial route has arguments and since its automatically pushed it has no way to resolve it's arguments.

In this case you should specify the initial route and pass the args to it as you push it
 
push(UserPageWrapperRoute(chicken:[UserDetailRoute(...args)]))

**Comment by hieubui2409** [comment]
So, assume, initial route will auto push without accepting args, so I should create an initial (such as empty page route) and push to the one with args?). I wonder if the same thing (not accept args) happen with redirect route or the child route with path `''` or not?
Thank you. @Milad-Akarie
